tidy3d/plugins/dispersion/fit_web.py

```diff
@@ -102,15 +136,55 @@ def _set_url(config_env: Literal["default", "dev", "prod", "local"] = "default")
 
         return URL_ENV[_env]
 
-    def fit(  # pylint:disable=arguments-differ, too-many-arguments, too-many-locals
+    @staticmethod
+    def _setup_server(url_server: str):
+        """set up web server access
+
+        Parameters
+        ----------
+        url_server : str
+            URL for the server
+        """
+
+        from ...web.auth import (  # pylint:disable=import-outside-toplevel, unused-import
+            get_credentials,
+        )
+        from ...web.httputils import (  # pylint:disable=import-outside-toplevel
+            get_headers,
+        )
+
+        # get_credentials()
+        access_token = get_headers()
+        headers = {"Authorization": access_token["Authorization"]}
+
+        # test connection
+        resp = requests.get(url_server + "/health")
+        try:
+            resp.raise_for_status()
+        except Exception as e:
+            log.error(e)
+            raise WebError(  # pylint:disable=raise-missing-from
+                "Connection to the server failed. Please try again."
+            )
+
+        # test authorization
+        resp = requests.get(url_server + "/health/access", headers=headers)
+        try:
+            resp.raise_for_status()
+        except Exception as e:
+            log.error(e)
+            raise WebError(  # pylint:disable=raise-missing-from
+                "Authorization to the server failed. Please try again."
+            )
```

Comment:
we can `raise WebError(...) from e` to include the original exception.

Reply:
revsied.